aws/resource_aws_budgets_budget.go

```diff
@@ -249,8 +422,47 @@ func resourceAwsBudgetsBudgetUpdate(d *schema.ResourceData, meta interface{}) er
 		return fmt.Errorf("update budget failed: %v", err)
 	}
 
+	err = resourceAwsBudgetsBudgetNotificationsUpdate(d, meta)
+
+	if err != nil {
+		return fmt.Errorf("update budget notification failed: %v", err)
+	}
+
 	return resourceAwsBudgetsBudgetRead(d, meta)
 }
+func resourceAwsBudgetsBudgetNotificationsUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AWSClient).budgetconn
+	accountID, budgetName, err := decodeBudgetsBudgetID(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	if d.HasChange("notification") {
+		o, n := d.GetChange("notification")
+		os := o.(*schema.Set)
+		ns := n.(*schema.Set)
+
+		removeNotifications, _ := expandBudgetNotificationsUnmarshal(os.Difference(ns).List())
+		addNotifications, addSubscribers := expandBudgetNotificationsUnmarshal(ns.Difference(os).List())
+
+		for _, notification := range removeNotifications {
+			client.DeleteNotification(&budgets.DeleteNotificationInput{
+				Notification: notification,
+				BudgetName:   aws.String(budgetName),
+				AccountId:    aws.String(accountID),
+			})
+		}
+
+		err = resourceAwsBudgetsBudgetNotificationsCreate(addNotifications, addSubscribers, budgetName, accountID, meta)
+
+		if err != nil {
+			return err
```

Comment:
We should return error context here for operators and code maintainers:

```suggestion
			return fmt.Errorf("error creating Budget (%s) Notifications: %s", d.Id(), err)
```